timesketch/api/v1/resources/event.py

```diff
@@ -1208,3 +1248,156 @@ def post(self, sketch_id):
+
+        Returns:
+            HTTP_STATUS_CODE_OK if successful
+
+        Raises:
+            HTTP_STATUS_CODE_NOT_FOUND: if sketch or event does not exist
+            HTTP_STATUS_CODE_BAD_REQUEST: if the request is malformed, e.g.
+                events or tags are not lists or the number of events or tags
+                exceeds the maximum allowed
+            HTTP_STATUS_CODE_FORBIDDEN: if the user does not have write access
+                to the sketch
+        """
+
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sketch found with this ID.")
+
+        if not sketch.has_permission(current_user, "write"):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN,
+                "User does not have write access controls on sketch.",
+            )
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        events = form.get("events", [])
+        if not isinstance(events, list):
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, "Events need to be a list")
+
+        if len(events) > self.MAX_EVENTS_TO_TAG:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Cannot untag more than {0:d} events in a single "
+                "request".format(self.MAX_EVENTS_TO_TAG),
+            )
+
+        searchindex_id = form.get("searchindex_id")
+        searchindex_name = form.get("searchindex_name")
+
+        if not (searchindex_id or searchindex_name):
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                "No search index information supplied.",
+            )
+
+        searchindex = None
+        # in both cases we are flexible, no matter what was supplied
+        if searchindex_name:
+            searchindex = SearchIndex.query.filter_by(
+                index_name=searchindex_name
+            ).first()
+        elif searchindex_id:
+            searchindex = SearchIndex.query.get(searchindex_id)
+
+        if not searchindex:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Unable to find the Search index.",
+            )
+
+        if searchindex.get_status.status == "deleted":
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Unable to query event on a closed search index.",
+            )
+
+        tags = form.get("tags", [])
+        if not isinstance(tags, list):
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, "Tags need to be a list")
+
+        if len(tags) > self.MAX_TAGS_PER_REQUEST:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Cannot untag more than {0:d} tags in a single "
+                "request".format(self.MAX_TAGS_PER_REQUEST),
+            )
+
+        datastore = self.datastore
+
+        for _event in events:
+            result = self.datastore.get_event(searchindex.index_name, _event.get("_id"))
```

Comment:
If I get this part right, it will not find any event if the event is not part of the provided searchindex.

Problem: You allow for 500 events, but just one searchindex_id/name. So what happens if the event is not in this one searchindex?

Possible solution: Don't provide a static searchindex_id/name parameter and just dynamically get the searchindex for each event via something like this: 
```
searchindex = SearchIndex.query.filter_by(
                index_name = _event.get("_index")
            ).first()
```

Reply:
yep changed it